### ISSUE TITLE
Repetitive error message on every keystroke

### ISSUE THREAD

**Issue by filipesilva** [Microsoft/vscode-tslint#168]
When editing a project with no `node_modules` and a `rulesDirectory` entry in `tslint.json` such as:
```
  "rulesDirectory": [
    "node_modules/codelyzer"
  ],
```

I'll get an error message on the top of VSCode on every keystroke:
```
vscode-tslint: Cannot read tslint configuration - 'Could not find custom rule directory: <edited out path>'
```

I understand why this error appears - the folder I listed does not exist.

But it's quite annoying to have this message constantly sliding down on every. single. keystroke. Pressing escape to clear the message does little to help, since it will just be there on the next keystroke anyway.

I'd like to request a way to turn it off, or maybe a long-ish timeout (like 5 minutes) before appearing again.

My usecase for not having this directory is that I often work on code generating blueprints and custom folder structures that will symlink `node_modules` at build time.

**Comment by egamma** [comment]
@filipesilva VS Code also supports workspace settings, have you tried to override the `rulesDirectory` setting in the workspace settings for directories that have no node_modules folder?

**Comment by filipesilva** [comment]
Heya @egamma, thank you for the swift reply. That works yes, but in doing so I lose it for the whole project... or is that incorrect? Is there a way to selectively say 'for this folder, override X'?

**Comment by filipesilva** [comment]
As a concrete example, the repo in question is https://github.com/angular/angular-cli.

For most of it, I do wish to have linting. But for in https://github.com/angular/angular-cli/tree/master/packages/%40angular/cli/blueprints/ng/files, linting will not work because that one also has a `tslint.json` which is part of the blueprint.

On every keystroke while editing those files, I get the error message sliding down and hiding my opened file list:

![image](https://cloud.githubusercontent.com/assets/4172079/23277734/de30acd8-fa06-11e6-8223-ea100dde11db.png)

**Comment by egamma** [comment]
I see the annoyance. 

The issue is that the error message comes from tslint directly and at the extension level I don't want to start parsing error messages from tslint and treat them differently based on string matching. One option would be to show all tslint messages in the tslint output window only and show in the tslint statusbar item that tslinting has failed, but I'm not there yet 😃. 

So here is another approach for you to try. when you always want to use 'codelyzer'  rules in your projects then install codelyzer globally and define the rulesDirectory *user setting* (not in tslint.json) so that it points to this global file using an absolute path. I think that the rulesDirectory definition in the vscode user settings settings wins over the rulesDirectory in the tslint.json, but I haven't tried it out.

**Comment by filipesilva** [comment]
I tried your tip, using ` "tslint.rulesDirectory": "C:\\Program Files\\nodejs\\node_modules\\codelyzer"` at the user settings and it didn't seem to make a difference. I think the local one did win over.

I understand this is both a rather niche problem and that it would introduce unnecessary complexity. I just wanted wanted to bring it up in case there was an obvious solution. 

Thanks for taking the time to help me out @egamma 👍

**Comment by egamma** [comment]
Sorry about that, then I suggest you go with my last option above, disable tslint using `tslint.enable: false` in the workspace settings for projects that do not have rules module.

**Comment by filipesilva** [comment]
Will do!

**Comment by egamma** [comment]
@filipesilva the fix to https://github.com/Microsoft/vscode-tslint/issues/197 should help you as well, the error message will be shown in the output window only and not in a message box.

The fix will be in tslint 0.1.13. So I'm closing this issue.

**Comment by filipesilva** [comment]
@egamma that's great news, thank you for taking the time to address this!
